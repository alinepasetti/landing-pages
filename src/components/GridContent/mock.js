export default {
  title: 'O t√≠tulo',
  html: `<h1>HTML Ipsum Presents</h1>
<p><strong>Pellentesque habitant morbi tristique</strong> senectus et netus et
malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae,
ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas
semper. <em>Aenean ultricies mi vitae est.</em></p>

<h2>Header Level 2</h2>

<ol>
	<li>Lorem ipsum dolor sit amet, consectetuer adipiscing elit.</li>
	<li>Aliquam tincidunt mauris eu risus.</li>
</ol>

<blockquote><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.
Vivamus magna. Cras in mi at felis aliquet congue. Ut a est eget ligula molestie
gravida. Curabitur massa. Donec eleifend, libero at sagittis mollis, tellus est
malesuada tellus, at luctus turpis elit sit amet quam. <a href='localhost:3000/'>Link</a>
Vivamus pretium ornare est.</p></blockquote>

<h3>Header Level 3</h3>

<ul>
	<li>Lorem ipsum dolor sit amet, consectetuer adipiscing elit.</li>
	<li>Aliquam tincidunt mauris eu risus.</li>
</ul>

<pre><code>
#header h1 a {
display: block;
width: 300px;
height: 80px;
}
</code></pre>
  `,
  background: false,
};
