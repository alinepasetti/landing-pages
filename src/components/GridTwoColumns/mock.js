export default {
  title: 'Grid two column',
  text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec a
  efficitur ex. Donec viverra neque quis velit convallis scelerisque.
  Etiam iaculis, ligula quis aliquet posuere, eros enim vestibulum risus,
  fermentum tempus augue lorem nec ex. Fusce fringilla nisl nec odio
  commodo accumsan. Proin placerat euismod quam, in convallis mauris
  eleifend pharetra. In sodales ultricies augue, nec finibus nisl aliquet
  et. Nam rhoncus quam non lobortis finibus`,
  srcImage: 'assets/images/javascript.svg',
};
