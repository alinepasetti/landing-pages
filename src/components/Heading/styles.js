import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`font-size: ${theme.fonts.sizes.medium};`,
  medium: (theme) => css`font-size: ${theme.fonts.sizes.large};`,
  large: (theme) => css`font-size: ${theme.fonts.sizes.xlarge};`,
  huge: (theme) => css`
  font-size: ${theme.fonts.sizes.xhuge};
  ${mediaFont(theme)}`,
};

const mediaFont = (theme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.fonts.sizes.xlarge};
  }`;

export const Title = styled.h1`
 ${({ theme, lightMode, uppercase, size }) => css`
 color: ${lightMode ? theme.colors.white : theme.colors.secondaryColor};
 text-transform: ${uppercase ? 'uppercase' : 'none'};
 ${titleSize[size](theme)}
 `}
`;
