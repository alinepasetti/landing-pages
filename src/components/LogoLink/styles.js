import styled, { css } from 'styled-components';

export const Container = styled.a` ${({ theme }) => css`
color: black;
text-decoration: none;
display: flex;
align-items: center;


> img {
  max-height: 3.5rem;
}
`}`;
