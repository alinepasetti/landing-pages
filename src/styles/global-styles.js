import { createGlobalStyle, css } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
html {
  /* this allows us to manage rem measure as if it was pixels */
  font-size: 62.5%;
}
body {
  font-size: 1.6rem;
  ${({ theme }) => css`
  background: ${theme.colors.mainColor};
  font-family: ${theme.fonts.default};
`}
}

h1, h2, h3, h4, h5, h6 {
  ${({ theme }) => css`
  font-family: ${theme.fonts.secondary};
  margin: ${({ theme }) => theme.spacings.large} 0;
`}
}

p {
  margin: ${({ theme }) => theme.spacings.medium} 0;
}

ul, ol {
  margin: ${({ theme }) => theme.spacings.medium};
  padding: ${({ theme }) => theme.spacings.medium};
}

a {
  color: ${({ theme }) => theme.colors.secondaryColor};
}
`;
