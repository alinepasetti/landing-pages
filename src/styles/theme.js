export const theme = {
  colors: {
    mainColor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
    mediumGray: '#DDDDDD',
  },
  fonts: {
    default: '"Open Sans", sans-serif',
    secondary: '"Montserrat", sans-serif;',
    sizes: {
      xsmall: '8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.6rem',
      xlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
  },
  media: {
    medium: '(max-width: 768px)',
  },
  spacings: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.6rem',
    xlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
};
