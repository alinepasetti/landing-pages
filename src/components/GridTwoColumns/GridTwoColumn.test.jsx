import { renderTheme } from '../../styles/render-theme';
import { screen } from '@testing-library/react';
import { GridTwoColumn } from '.';
import mock from './mock';

describe('<GridTwoColumn />', () => {
  it('should render', () => {
    const { container } = renderTheme(<GridTwoColumn {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
