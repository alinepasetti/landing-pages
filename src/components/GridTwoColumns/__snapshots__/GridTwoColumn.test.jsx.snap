// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> should render 1`] = `
.c5 {
  color: #dc143c;
  text-transform: uppercase;
  font-size: 6.4rem;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.6rem;
}

.c2 .c4 {
  margin-bottom: 4.8rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.6rem;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.8rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.6rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h1
            class="c4 c5"
          >
            Grid two column
          </h1>
          <p
            class="c6"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec a
  efficitur ex. Donec viverra neque quis velit convallis scelerisque.
  Etiam iaculis, ligula quis aliquet posuere, eros enim vestibulum risus,
  fermentum tempus augue lorem nec ex. Fusce fringilla nisl nec odio
  commodo accumsan. Proin placerat euismod quam, in convallis mauris
  eleifend pharetra. In sodales ultricies augue, nec finibus nisl aliquet
  et. Nam rhoncus quam non lobortis finibus
          </p>
        </div>
        <div
          class=""
        >
          <img
            alt="Grid two column"
            class=""
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
